src/core/public/entry_point.ts

```diff
@@ -37,11 +37,18 @@ const injectedMetadata = JSON.parse(
  * `apmConfig` would be populated with relavant APM RUM agent
  * configuration if server is started with `ELASTIC_APM_ACTIVE=true`
  */
-if (process.env.IS_KIBANA_DISTRIBUTABLE !== 'true' && injectedMetadata.vars.apmConfig != null) {
+const APM_ENABLED =
+  process.env.IS_KIBANA_DISTRIBUTABLE !== 'true' && injectedMetadata.vars.apmConfig != null;
+
+if (APM_ENABLED) {
   // @ts-ignore
   // eslint-disable-next-line @typescript-eslint/no-var-requires
-  const { init } = require('@elastic/apm-rum');
-  init(injectedMetadata.vars.apmConfig);
+  const { init, apm } = require('@elastic/apm-rum');
+  const config = injectedMetadata.vars.apmConfig;
+  if (config.globalLabels) {
+    apm.addLabels(config.globalLabels);
```

Comment:
would let user to search the APM trace using `kibana_uuid` 

https://github.com/elastic/kibana/blob/57345e092e72a76d3180cdb85658956da944bb9d/src/apm.js#L68

Reply:
Already present before this PR, but

https://github.com/elastic/kibana/blob/bf04235dae35452061cc7ea3d86d96c19a58206c/src/apm.js#L67

Is potentially not accessing the `uuid` file from the correct location, as the `data` folder is configurable via `path.data`, so the file may be in a totally other folder than `join(ROOT_DIR, 'data')`

See 

https://github.com/elastic/kibana/blob/d0aeadf13eb8f7621d21f41eae480f011a561e64/src/core/server/path/index.ts#L64-L69

and usage in `src/core/server/uuid/resolve_uuid.ts` for instance.

Reply:
I see your point, But do we need to account for the `path.data` config for APM as its purely used only for dev environment and not for distributable. Also, I am not sure if there is a nice way of accessing the path config from `kibana.yml` in the `apm.js` file as we need it to run before the app code starts executing. Let me know if you have any ideas.